[Quickstart] Added server side logout Obj-c

benlizar opened the pull request:
<!---
Notes:
- Your PR should conform to our [Contributing Guidelines](https://github.com/auth0/docs/blob/master/CONTRIBUTING.md)
- If applicable, add details to the [update feed](https://github.com/auth0/docs/tree/master/updates)
- Make sure your PR gets deployed to a Heroku [Review App](https://github.com/auth0/docs/blob/master/CONTRIBUTING.md#review-apps) without errors
- Please include a short description
- If your PR is a work in progress, title it [DO NOT MERGE]
--->

Commit:
Added snippet for logout

Commit:
Added return value

lbalmaceda commented on articles/quickstart/native/ios-objc/_includes/_login_centralized.md:
```diff
@@ -114,6 +114,27 @@ To learn more about the `credentials` object, read the [Credentials](https://git
+    func logOutUser(callback: @escaping(Bool, String) -> Void){
+        Auth0
+            .webAuth()
+            .clearSession(federated: true){
```
I don't think this parameter is necessary. Can you remove it please?

benlizar replied:
@lbalmaceda the parameter is required 
![image](https://user-images.githubusercontent.com/36707032/63733328-1b65f780-c835-11e9-865f-497daa7ab122.png)

albertoperdomo replied:
If it's required please set to false, which is what we are doing across all QS @benlizar (default behavior).

lbalmaceda commented on articles/quickstart/native/ios-objc/_includes/_login_centralized.md:
````diff
@@ -114,6 +114,27 @@ To learn more about the `credentials` object, read the [Credentials](https://git
+            }
+    }
+```
+After the logout, the callback will return a BOOL with the logout status.
````
```suggestion
After the call, the callback will receive a BOOL with the logout status.
```

lbalmaceda commented on articles/quickstart/native/ios-objc/_includes/_login_centralized.md:
```diff
@@ -114,6 +114,27 @@ To learn more about the `credentials` object, read the [Credentials](https://git
 
 <%= include('../../../../_includes/_logout_url') %>
 
+To clear on server side the session add the following snipped to the HybridAuth.swift file:
```
The file is already clarified on the snippet and we can remove it👍 

```suggestion
To clear the session on the server side you need to invoke the `clearSession` method. Add the following snippet:
```

lbalmaceda commented on articles/quickstart/native/ios-objc/_includes/_login_centralized.md:
````diff
@@ -114,6 +114,27 @@ To learn more about the `credentials` object, read the [Credentials](https://git
+```swift
+// HybridAuth.swift
+
+    @objc
````
See if you can remove all these spaces on the left so that the **starting** indentation matches the `// HybridAuth.swift` line and so, all the snippets above.

lbalmaceda reviewed (changes requested):
Thanks benja! Left some comments

benlizar committed:
Update articles/quickstart/native/ios-objc/_includes/_login_centralized.md

Co-Authored-By: Luciano Balmaceda <balmacedaluciano@gmail.com>

benlizar committed:
Update _login_centralized.md

benlizar committed:
Update _login_centralized.md

benlizar committed:
Update articles/quickstart/native/ios-objc/_includes/_login_centralized.md

Co-Authored-By: Luciano Balmaceda <balmacedaluciano@gmail.com>

benlizar reviewed (commented):
Fixed and corrected all suggestions

damieng commented:
@cocojoe I wonder if we should make clearSessions federated parameter default to false?

https://github.com/auth0/Auth0.swift/blob/34b9258412184faa8b9e81d21c51f8cb5c0d12ec/Auth0/WebAuth.swift#L234 indicates it should but the definition at doesn't make it so https://github.com/auth0/Auth0.swift/blob/34b9258412184faa8b9e81d21c51f8cb5c0d12ec/Auth0/WebAuth.swift#L248

FWIW Auth0.Net's logout specifying federated is optional and defaults to false.

benlizar committed:
Update _login_centralized.md

Commit:
Merge remote-tracking branch 'origin' into Reviewing-logout-objc

Commit:
cleared snniped code

Commit:
Merge branch 'Reviewing-logout-objc' of https://github.com/auth0/docs into Reviewing-logout-objc

Commit:
added logout callback

- Changed user-sessions to add logout urls
- Added logout callback on download modal
- Added callback snippet in 00 login

Commit:
Removed the callback note
